src/navigation/drive.py

```diff
@@ -1,63 +1,160 @@
-from typing import Tuple
-
+from typing import List, Tuple, ClassVar
+from dataclasses import dataclass
 import numpy as np
+from shapely.geometry import Polygon, Point
 
 from geometry_msgs.msg import Twist
 from util.SE3 import SE3
+
 from util.np_utils import angle_to_rotate
+from failure_zone import FailureZone
+from path_planner import PathPlanner
 
 MAX_DRIVING_EFFORT = 1
 MIN_DRIVING_EFFORT = -1
 TURNING_P = 10.0
+ROVER_WIDTH = 1  # meters
+
+
+class Driver:
+    planner: PathPlanner
+
+    def __init__(self) -> None:
+        self.planner = PathPlanner()
+
+    def add_failure_zone(self, failure_zone: FailureZone) -> None:
+        """
+        Add a newly-detected failure zone to the PathPlanner.
+        """
+        self.planner.add_failure_zone(failure_zone)
+    
+    def add_post_as_failure_zone(self, post_pos: np.ndarray) -> None:
+        """
+        Add a newly-detected post as a failure zone to the PathPlanner.
```

Comment:
we should only add it after we've driven to it so the comment isn't exactly right

Reply:
amended comment